mlir/lib/Dialect/MIOpen/Tuning/CMakeLists.txt

```diff
@@ -9,6 +9,8 @@ add_mlir_dialect_library(MLIRMIOpenTuning
 
   DEPENDS
   MLIRSupport
+  MLIRGPUOpInterfacesIncGen
```

Comment:
I'd be surprised to see `MLIRGPUOpInterfacesIncGen` is necessary inside `MLIRMIOpenTuning` library. This targets use no gpu ops. Could you double check whether/why this is needed?

Reply:
I think it's to do with the GPU async interface?

Reply:
`MLIRMIOpenTuning` Shouldn't use any gpu async stuff? It only derived tuning parameters based on conv/grid level ops.

Reply:
It might have dependencies that do and that don't declare their dependencies correctly?

If nothing else, `#include`ing the GPU dialect somewhere, even if we don't actually use it in that file anymore, could do this.

We might want to go hunting for redundant includes

Reply:
> It might have dependencies that do and that don't declare their dependencies correctly?

That's sounds to be the case if this is truly needed to make compilation pass. And if so I'd prefer this PR to address the redundant `#include` from `MLIRMIOpenTuning` component only, such that we don't have to pull in gpu ops dependencies. Otherwise we'd be creating new problem out from the fixing of the old one.

Reply:
Yeah, good idea